enumerable_methods.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while self[i]
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_each { |e| yield(index(e), e) }
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_items = []
+    my_each { |item| selected_items << item if yield(item) }
+    selected_items
+  end
+
+  def my_all?
+    my_each { |item| return false unless yield(item) }
+    true
+  end
```

Comment:
What happens when a block is not given?

`[1, true, :sym].my_all?` ==  `[1, true, :sym].all?` == `true`

`[1, 2, 3].my_all?(Integer)` ==  `[1, 2, 3].all?(Integer)` == `true`

However, you method throws an error:

> ArgumentError:
       wrong number of arguments (given 1, expected 0)

I advise you to test all your method against the once ruby provide.